src/kubectlUtils.ts

```diff
@@ -151,16 +167,18 @@ export async function getPodSelector(resource: string, kubectl: Kubectl): Promis
     });
 }
 
-export async function getPods(kubectl: Kubectl, selector: any): Promise<PodInfo[]> {
-    if (!selector) {
-        return [];
+export async function getPods(kubectl: Kubectl, selector: any, namespace: string = null): Promise<PodInfo[]> {
+    let ns = namespace;
+    if (!ns) {
+        ns = await currentNamespace(kubectl);
+    }
+    let nsFlag = `--namespace=${ns}`
+    if (ns === 'all') {

```

Comment:
Nit: Is `all` a legal name for a Kubernetes namespace?  It looks like this would do the wrong thing if someone was using a namespace with this name... unlikely as it may seem!

Reply:
Do you have a better magic value?

Reply:
The magic string doesn't need to be a valid namespace name does it?  So it could be something like `*all*` perhaps (assuming namespaces need to be DNS-valid?)?